test/internet_connection_test.dart

```diff
@@ -21,6 +23,55 @@ void main() {
       expect(await checker.hasInternetAccess, false);
     });
 
+    test('hasInternetAccess invoke responseStatusFn to determine success',
+        () async {
+      await TestHttpClient.run((client) async {
+        client.responseBuilder =
+            (req) => TestHttpClient.createResponse(statusCode: 500);
+        const expectedStatus = true;
+        final checker = InternetConnection.createInstance(
+          customCheckOptions: [
+            InternetCheckOption(
+              uri: Uri.parse('https://www.example.com/nonexistent-page'),
+              responseStatusFn: (response) => expectedStatus,
+            ),
+          ],
+          useDefaultOptions: false,
+        );
+
+        expect(await checker.hasInternetAccess, expectedStatus);
+      });
+    });
+
+    test('hasInternetAccess send custom header on request', () async {
+      await TestHttpClient.run((client) async {
+        const expectedStatus = true;
+        const expectedHeaders = {'Authorization': 'Bearer token'};
+
+        client.responseBuilder = (req) {
+          for (final header in expectedHeaders.entries) {
+            final key = header.key;
+            if (!req.headers.containsKey(key) ||
+                req.headers[key] != header.value) {
+              return TestHttpClient.createResponse(statusCode: 500);
+            }
+          }
+          return TestHttpClient.createResponse(statusCode: 200);
+        };
```

Comment:
Why is this code necessary? The tests are running properly without these additional Mock codes.

Reply:
It is use to verify our expected header was exist on the request.